### PR TITLE
Mmahbubu/provenance fix grpah replace

### PR DESCRIPTION
**Pull request by tachyon77**
Fixes provenance algorithm when subgraphs are replaced.
Unit test change explains what changed.

### CODE REVIEW

File: src/ngraph/graph_util.cpp [hunk 1]
```diff
@@ -153,14 +153,20 @@ void ngraph::replace_node(std::shared_ptr<Node> target,
     {
         auto common_args = ngraph::find_common_args(target, replacement);
 
-        auto set_replacement_prov = [replacement](std::shared_ptr<Node> node) {
-            replacement->merge_provenance_tags_from(node);
+        std::unordered_set<string> removed_subgraph_tags;
```
Comment (diyessi) [inline src/ngraph/graph_util.cpp]: Use an ordered set so that results are consistent on different runs.

File: src/ngraph/node.cpp [hunk 2]
```diff
@@ -328,6 +328,14 @@ void Node::add_provenance_tag(const std::string& tag)
     m_provenance_tags.insert(tag);
 }
 
+void Node::add_provenance_tags(const std::unordered_set<std::string>& tag_set)
```
Comment (diyessi) [inline src/ngraph/node.cpp]: ordered set

File: src/ngraph/node.hpp [hunk 3]
```diff
@@ -368,6 +368,7 @@ namespace ngraph
 
         const std::unordered_set<std::string>& get_provenance_tags() const;
         void add_provenance_tag(const std::string& tag);
+        void add_provenance_tags(const std::unordered_set<std::string>& tag_set);
```
Comment (diyessi) [inline src/ngraph/node.hpp]: ordered set

File: src/ngraph/graph_util.cpp (hunk 1 above)
Comment (tachyon77) [reply src/ngraph/graph_util.cpp]: @diyessi Provenance tags (m_provenance_tags) is unordered. I am not sure making this one sorted would be enough.

**Pull request merged**